onnxruntime/core/optimizer/constant_folding.h

```diff
@@ -18,9 +20,9 @@ class ConstantFolding : public GraphTransformer {
  public:
   /** Constant folding will not be applied to nodes that have one of initializers from excluded_initializers as input.
       For pre-training, the trainable weights are those initializers to be excluded. */
-  ConstantFolding(const std::unordered_set<std::string>& compatible_execution_providers = {},
-                  const std::unordered_set<std::string>& excluded_initializers = {}) noexcept
-      : GraphTransformer("ConstantFolding", compatible_execution_providers), excluded_initializers_(excluded_initializers) {}
+  ConstantFolding(const IExecutionProvider* cpu_execution_provider,
+                  const std::unordered_set<std::string>& compatible_execution_providers = {},
```

Comment:
cpu_execution_provider should be a reference as you require a non nullptr value.